### DISCUSSION TITLE
i3wm on Manjaro Linux

### DISCUSSION THREAD

**Discussion by apastuszak** [i3/i3#4567] [Q&A] [answered]
I have a laptop with Manjaro Linux installed that has KDE 5 on it. Last week I installed i3 on it and got hooked pretty fast. Now I'm trying to set up i3 and am running into two odd issues:

1. I installed nm-applet in order to get wifi working. And it's "iffy" at best. I can't get nm-applet to save my wifi password, so every time I login, I get prompted for my wifi password.
2. And when I launch the Nextcloud client, I also need to login to Nextcloud every time I login to the laptop.

Assuming this was an issue with KWallet, I went into my i3 config and added the following line:

exec --no-startup-id /usr/lib/pam_kwallet_init to get kwallet to unlock when I login

That didn't help. I launched KWallet and confirmed it was unlocked and left it running, and both apps still want me to login.

How do I get these apps to store passwords, so I am not asked to login every time I launch the apps?

**Comment by apastuszak** [comment]
Ok, looks like I got around the NeworkManager.

I ran nmtui as root:

sudo nmtui

and had it save my password and told it to remember for all users. I am no longer being prompted for my WiFi password.

Still need to fix the NextCloud issue.

**Comment by apastuszak** [comment]
Looks like I was wrong.  That survived a log out and login, but failed after a reboot. I was prompted for my WiFi password again.

**Comment by dojero** [comment]
I wonder why you consider this an I3 problem. Why not just make Network Manager save your password, as this seems to suggest is possible:
https://unix.stackexchange.com/questions/587098/networkmanager-does-not-save-wifi-passwords

-------- Original Message --------

**Reply by apastuszak** [reply]
It took some effort to figure this out, but I finally got NetworkManager to save my password.

But the NextCloud problem is still there. It's not an i3 "problem." This obviously works for other people. I just want to know what other people are doing to get their Nextcloud client to work, because it works just fine in KDE or Gnome.

**Comment by apastuszak** [comment, marked as answer]
Ok, I've figured it out. With some experimenting and the Arch wiki, I was able to figure it out.

In order to get Nextcloud to save your password, you will need to have the following installed:

qtkeychain-qt5
libsecrets
gnome-keyring

Once I had all of these installed and I logged into Nextcloud, it prompted me to create a password for my new keyring and my Nextcloud authentication was saved. This may work with NetworkManager too, but I haven't tried it.

According to the Arch wiki, you need to be running something that provides org.freedesktop.secrets, which kwallet does not natively provide. The only two apps that do in the Arch repositories is gnome-keyring and keepassxc.

Hopefully this will be useful for someone else.
